src/instrument.ts

```diff
@@ -165,11 +176,20 @@ export function startTrace<T>(
         }
       }
 
-      if (config.that) {
-        const fn = cb.bind(config.that);
-        return fn(span);
-      } else {
-        return cb(span);
+      // If at all the invoked function throws an exception,
+      // record the exception and then end this span.
+      try {
+        if (config.that) {
```

Comment:
Have we even used "that" property of "traceconfig" anywhere? What is even the use of this ?

Reply:
It was to capture a scenario of the this object being used to capture the context of where it was run in case the caller did not use an arrow function `startTrace(name, config) => span {` and instead passed in a function `startTrace(name, config, function(span) {`.

Reply:
I don't see reference of it anywhere. If are not using it, can we remove ?

Reply:
Sure!